packages/back-end/src/models/FeatureModel.ts

```diff
@@ -678,7 +687,13 @@ export async function editFeatureRule(
     ...changes.rules[environment][i],
     ...updates,
   } as FeatureRule;
-
+  if (
+    (revision.status === "approved" ||
+      revision.status === "changes-requested") &&
+    resetReview
+  ) {
+    changes.status = "pending-review";
+  }
```

Comment:
There are lots of ways to edit a draft that should reset a review.  I think we're missing a bunch, for example `setDefaultValue`.  I think it would be safer to just add a `resetReview` required argument to `updateRevision`.  That will do 2 things:

1.  Abstract this logic of checking statuses and resetting it so we don't have to repeat it throughout the codebase
2.  Leverage typescript to force us to update every instance

For things like `putRevisionComment` that shouldn't reset anything, we can just hardcode `false` when we update the revision.